zulip/integrations/log2zulip/log2zulip

```diff
@@ -67,47 +66,53 @@ def process_lines(raw_lines, file_name):
 def process_logs():
     # type: () -> None
     for filename in log_files:
-        data_file_path = "/var/tmp/log2zulip.state"
+        data_file_path = os.path.join(tempfile.gettempdir(), 'log2zulip.state')
     mkdir_p(os.path.dirname(data_file_path))
     if not os.path.exists(data_file_path):
-        open(data_file_path, "w").write("{}")
-    last_data = ujson.loads(open(data_file_path).read())
+        open(data_file_path, 'w').write('{}')
+    last_data = json.loads(open(data_file_path).read())
     new_data = {}
     for log_file in log_files:
         file_data = last_data.get(log_file, {})
         if not os.path.exists(log_file):
             # If the file doesn't exist, log an error and then move on to the next file
             print("Log file does not exist or could not stat log file: %s" % (log_file,))
             continue
-        length = int(subprocess.check_output(["wc", "-l", log_file]).split()[0])
-        if file_data.get("last") is None:
-            file_data["last"] = 1
-        if length + 1 < file_data["last"]:
+        length = int(subprocess.check_output(['wc', '-l', log_file]).split()[0])
+        if file_data.get('last') is None:
+            file_data['last'] = 1
+        if length + 1 < file_data['last']:
             # The log file was rotated, restart from empty.  Note that
             # because we don't actually store the log file content, if
             # a log file ends up at the same line length as before
             # immediately after rotation, this tool won't notice.
-            file_data["last"] = 1
-        output = subprocess.check_output(["tail", "-n+%s" % (file_data["last"],), log_file])
+            file_data['last'] = 1
+        output = subprocess.check_output(['tail', '-n+%s' % (file_data['last'],), log_file])
         new_lines = output.decode('utf-8', errors='replace').split('\n')[:-1]
         if len(new_lines) > 0:
             process_lines(new_lines, filename)
-            file_data["last"] += len(new_lines)
+            file_data['last'] += len(new_lines)
         new_data[log_file] = file_data
-    open(data_file_path, "w").write(ujson.dumps(new_data))
+    open(data_file_path, 'w').write(json.dumps(new_data))
 
-if __name__ == "__main__":
+if __name__ == '__main__':
+    parser = zulip.add_default_arguments(argparse.ArgumentParser())  # type: argparse.ArgumentParser3
+    parser.add_argument('--control-path', default='/etc/log2zulip.conf')
+    args = parser.parse_args()
+    if not args.zulip_config_file and os.name == 'posix':
+        args.zulip_config_file = '/etc/log2zulip.zuliprc'
```

Comment:
Do you think this might be cleaner done by modifying `add_default_arguments` to support a `default_config_file_path` argument?

Reply:
It already does, right? `zulip_config_file` is the parameter for `--config-file`.

Reply:
So what I tried to do here is to use the default values for the `zuliprc` in any case possible: Only if we are on a posix system, the path should default to `/etc/log2zulip.zuliprc`, so that folks updating this script don't have to reconfigure it? Is that what you meant?